internals/secrethub/secret_reader.go

```diff
@@ -58,3 +61,23 @@ func (sr *bufferedSecretReader) ReadSecret(path string) (string, error) {
 func (sr bufferedSecretReader) SecretsRead() []string {
 	return sr.secretsRead
 }
+
+type ignoreMissingSecretReader struct {
+	secretReader tpl.SecretReader
+}
+
+func newIgnoreMissingSecretReader(sr tpl.SecretReader) *ignoreMissingSecretReader {
+	return &ignoreMissingSecretReader{
+		secretReader: sr,
+	}
+}
+
+// ReadSecret uses the underlying secret reader to read the secret, but ignores
+// errors for non-existing secrets. Instead, it returns the empty string.
+func (sr *ignoreMissingSecretReader) ReadSecret(path string) (string, error) {
+	secret, err := sr.secretReader.ReadSecret(path)
+	if err == api.ErrSecretNotFound {
```

Comment:
Question: what happens when say the repo doesn't exist? Does the server return `api.ErrSecretNotFound`? And by extension, how does this impact our ability to correctly ignore missing secrets? E.g. do we want to still ignore this when a different error (such as `api.ErrRepoNotFound`) is returned?

Consider whether we **can and want to** check the status code `404` instead of the exact error type to reduce coupling between the API implementation and the CLI.

Reply:
You are right. The server indeed returns a different error when the repo does not exists.
I'd say this error should also be ignored when the flag is used.

As for whether this is possible: I've opened https://github.com/secrethub/secrethub-go/pull/96 to add functionality to support this.

Reply:
Cool, approved that PR. You decide wether to hold on merging this PR or change this later.

Reply:
I've added the helper function in this PR, so that it isn't blocked by a release of secrethub-go. We can update it later to use the function of secrethub-go, once that is released.